diff --git a/arch/arm/include/asm/arch-mx6/imx-regs.h b/arch/arm/include/asm/arch-mx6/imx-regs.h
index a69a753..c7e289f 100644
--- a/arch/arm/include/asm/arch-mx6/imx-regs.h
+++ b/arch/arm/include/asm/arch-mx6/imx-regs.h
@@ -83,7 +83,7 @@
 #define MMDC1_ARB_END_ADDR              0xFFFFFFFF
 #else
 #define MMDC0_ARB_BASE_ADDR             0x10000000
-#define MMDC0_ARB_END_ADDR              0x7FFFFFFF
+#define MMDC0_ARB_END_ADDR              0x8FFFFFFF
 #define MMDC1_ARB_BASE_ADDR             0x80000000
 #define MMDC1_ARB_END_ADDR              0xFFFFFFFF
 #endif
diff --git a/board/freescale/imx/ddr/mx6q_4x_mt41j128.cfg b/board/freescale/imx/ddr/mx6q_4x_mt41j128.cfg
index bb6c60b..4737c9d 100644
--- a/board/freescale/imx/ddr/mx6q_4x_mt41j128.cfg
+++ b/board/freescale/imx/ddr/mx6q_4x_mt41j128.cfg
@@ -101,8 +101,8 @@ DATA 4 0x021b002c 0x000026D2
 DATA 4 0x021b0030 0x005A1023
 DATA 4 0x021b0008 0x09444040
 DATA 4 0x021b0004 0x00025576
-DATA 4 0x021b0040 0x00000027
-DATA 4 0x021b0000 0x831A0000
+DATA 4 0x021b0040 0x00000047
+DATA 4 0x021b0000 0x841A0000
 
 DATA 4 0x021b001c 0x04088032
 DATA 4 0x021b001c 0x0408803A
diff --git a/boards.cfg b/boards.cfg
index dbee7f6..8e6def3 100644
--- a/boards.cfg
+++ b/boards.cfg
@@ -328,6 +328,7 @@ Active  arm         armv7          mx6         freescale       mx6qsabreauto
 Active  arm         armv7          mx6         freescale       mx6qsabreauto       mx6dlsabreauto                        mx6qsabreauto:IMX_CONFIG=board/freescale/mx6qsabreauto/mx6dl.cfg,MX6DL            Fabio Estevam <fabio.estevam@freescale.com>
 Active  arm         armv7          mx6         freescale       mx6sabresd          mx6dlsabresd                          mx6sabresd:IMX_CONFIG=board/boundary/nitrogen6x/nitrogen6dl.cfg,MX6DL                                                             Fabio Estevam <fabio.estevam@freescale.com>
 Active  arm         armv7          mx6         freescale       mx6sabresd          mx6qsabresd                           mx6sabresd:IMX_CONFIG=board/freescale/imx/ddr/mx6q_4x_mt41j128.cfg,MX6Q                                                           Fabio Estevam <fabio.estevam@freescale.com>
+Active  arm         armv7          mx6         freescale       mx6sabresd          tbs2910                               mx6sabresd:IMX_CONFIG=board/freescale/imx/ddr/mx6q_4x_mt41j128.cfg,MX6Q,DDR_MB=2048
 Active  arm         armv7          mx6         freescale       mx6slevk            mx6slevk                              mx6slevk:IMX_CONFIG=board/freescale/mx6slevk/imximage.cfg,MX6SL                                                                   Fabio Estevam <fabio.estevam@freescale.com>
 Active  arm         armv7          mx6         gateworks       gw_ventana          gwventana                            gw_ventana:IMX_CONFIG=board/gateworks/gw_ventana/gw_ventana.cfg,MX6QDL,SPL                                                  Tim Harvey <tharvey@gateworks.com>
 Active  arm         armv7          mx6         solidrun        hummingboard        hummingboard_solo                     hummingboard:IMX_CONFIG=board/solidrun/hummingboard/solo.cfg,MX6S,DDR_MB=512                                                      Jon Nettleton <jon.nettleton@gmail.com>
diff --git a/include/configs/mx6sabresd.h b/include/configs/mx6sabresd.h
index e666ebb..4c52348 100644
--- a/include/configs/mx6sabresd.h
+++ b/include/configs/mx6sabresd.h
@@ -71,4 +71,6 @@
 #define CONFIG_POWER_PFUZE100
 #define CONFIG_POWER_PFUZE100_I2C_ADDR	0x08
 
+#include "tbs2910.h"
+
 #endif                         /* __MX6QSABRESD_CONFIG_H */
diff --git a/include/configs/tbs2910.h b/include/configs/tbs2910.h
new file mode 100644
index 0000000..2d66ea4
--- /dev/null
+++ b/include/configs/tbs2910.h
@@ -0,0 +1,115 @@
+
+#undef PHYS_SDRAM_SIZE
+#define PHYS_SDRAM_SIZE                (2u * 1024 * 1024 * 1024)
+
+/* USB Configs */
+#define CONFIG_CMD_USB
+#define CONFIG_USB_EHCI
+#define CONFIG_USB_EHCI_MX6
+#define CONFIG_USB_STORAGE
+#define CONFIG_USB_MAX_CONTROLLER_COUNT 2
+#define CONFIG_EHCI_HCD_INIT_AFTER_RESET        /* For OTG port */
+#define CONFIG_MXC_USB_PORTSC   (PORT_PTS_UTMI | PORT_PTS_PTW)
+#define CONFIG_MXC_USB_FLAGS    0
+#define CONFIG_USB_KEYBOARD
+#define CONFIG_SYS_USB_EVENT_POLL_VIA_CONTROL_EP
+
+#define CONFIG_CMD_HDMIDETECT
+
+#undef CONFIG_SYS_MMC_ENV_DEV
+#define CONFIG_SYS_MMC_ENV_DEV 2
+#define CONFIG_SYS_MMC_ENV_PART 1
+
+/* Boot menu */
+#define CONFIG_CMD_BOOTMENU
+#define CONFIG_MENU
+#define CONFIG_AUTOBOOT_KEYED
+#define CONFIG_MENU_SHOW
+
+#undef CONFIG_SYS_CBSIZE
+#define CONFIG_SYS_CBSIZE 1024
+
+#define CONFIG_SYS_PROMPT               "U-Boot=> "
+#define CONFIG_FIT
+
+//#define CONFIG_ENV_VARS_UBOOT_CONFIG
+
+#undef CONFIG_BOOTCOMMAND
+#define CONFIG_BOOTCOMMAND \
+	"mw ${loadaddr} 0x00 1024; " \
+	"run loader; " \
+	"usb stop" \
+	"source; " \
+	"usb start" \
+	"echo ${target} not found!; " \
+	"printenv"
+
+#undef CONFIG_EXTRA_ENV_SETTINGS
+#define CONFIG_EXTRA_ENV_SETTINGS \
+	ENV_SCAN_SETTINGS \
+	ENV_MISC_SETTINGS \
+	ENV_BOOTMENU_SETTINGS \
+	FILE_LOADER
+
+#define CONFIG_CMD_EXT4
+#define CONFIG_EFI_PARTITION
+
+#undef CONFIG_FEC_MXC_PHYADDR
+#define CONFIG_FEC_MXC_PHYADDR 4
+
+#undef CONFIG_BOOTDELAY
+#define CONFIG_BOOTDELAY 4
+
+#undef CONFIG_LOADADDR
+#define CONFIG_LOADADDR 0x18000000
+
+#undef EMMC_ENV
+
+//Console setup
+#define CONFIG_CONSOLE_MUX
+#define CONFIG_SYS_CONSOLE_IS_IN_ENV
+#define CONFIG_NETCONSOLE
+
+#define ENV_MISC_SETTINGS \
+       "fdt_high=0xffffffff\0"
+
+#define ENV_BOOTMENU_SETTINGS \
+       "itemA=echo 1 chosen!\0" \
+       "itemB=echo 2 chosen!\0" \
+       "itemC=echo 3 chosen!\0" \
+       "itemD=echo 4 chosen!\0"
+
+#define CONFIG_PREBOOT \
+       "setenv autoload no; " \
+       "usb start; " \
+       "mmc rescan; " \
+       "if dhcp; then setenv nc ,nc; fi; " \
+       "setenv stdin serial${nc},vga,usbkbd; " \
+       "setenv stdout serial${nc},vga; " \
+       "setenv stderr serial${nc},vga; " \
+       "setenv bootmenu_0 Default: Boot 1.=run itemA;" \
+       "setenv bootmenu_1 \"Boot 1.=setenv bootmenu_0 Default: Boot 1.=run itemA; saveenv; run itemA;\";" \
+       "setenv bootmenu_2 \"Boot 2.=setenv bootmenu_0 Default: Boot 2.=run itemB; saveenv; run itemB;\";" \
+       "setenv bootmenu_3 \"Boot 3.=setenv bootmenu_0 Default: Boot 3.=run itemC; saveenv; run itemC;\";" \
+       "setenv bootmenu_4 \"Boot 4.=setenv bootmenu_0 Default: Boot 4.=run itemD; saveenv; run itemD;\";"
+
+#define ENV_SCAN_SETTINGS \
+        "target=boot.scr\0" \
+        "busses_to_scan=tftp usb mmc\0" \
+        "filesystems_to_scan=ext4 fat\0" \
+        "devices_to_scan=0 1 2 3\0" \
+        "device_regions_to_scan=0 1 2 3\0" \
+        "fs_partitions_to_scan=1 2 3\0"
+
+#define FILE_LOADER \
+       "loader=echo Scanning for ${target}...;" \
+        "for dtype in ${busses_to_scan}; do " \
+                "if test ${dtype} = tftp && test ${nc} = ,nc; then " \
+                        "echo ${dtype}; " \
+                        "if ${dtype} ${target}; then " \
+                                "echo ${target} found!; " \
+                        "fi; " \
+                "fi; " \
+        "done\0"
+
+
